KGHM/main.js

```diff
@@ -1,6 +1,106 @@
-import './style.css'
+import hangmanImg1 from "./public/assets/1.png";
+import hangmanImg2 from "./public/assets/2.png";
+import hangmanImg3 from "./public/assets/3.png";
+import hangmanImg4 from "./public/assets/4.png";
+import hangmanImg5 from "./public/assets/5.png";
+import hangmanImg6 from "./public/assets/6.png";
+import hangmanImg7 from "./public/assets/7.png";
+import hangmanImg8 from "./public/assets/8.png";
+import hangmanImg9 from "./public/assets/9.png";
+import hangmanImg10 from "./public/assets/10.png";
+import hangmanImg11 from "./public/assets/fire.png";
 
-document.querySelector('#app').innerHTML = `
-  <h1>Hello Vite!</h1>
-  <a href="https://vitejs.dev/guide/features.html" target="_blank">Documentation</a>
-`
+const $ = (selector) => document.querySelector(selector);
+const $$ = (selector) => document.querySelectorAll(selector);
+
+const lifeArr = $$(".life-wrapper"); //목숨
+const wordArr = $(".words"); //단어ul
+const hangmanImg = $(".hangman-img"); //이미지
+const failModal = $(".fail-modal"); //실패 모달
+const successModal = $(".success-modal"); //실패 모달
+
+let answerWord = ""; //정답단어
+let chance = 10; // 카운트는 10
+let imageCount = 0; // 이미지 번째 수
+let status = ""; //성공, 실패
+
+const imageList = [
+  hangmanImg1,
+  hangmanImg2,
+  hangmanImg3,
+  hangmanImg4,
+  hangmanImg5,
+  hangmanImg6,
+  hangmanImg7,
+  hangmanImg8,
+  hangmanImg9,
+  hangmanImg10,
+  hangmanImg11,
+];
+
+//단어받기 통신
+async function getWord() {
+  const response = await fetch("http://puzzle.mead.io/puzzle?wordCount=1");
+  if (response.status === 200) {
+    const data = await response.json();
+    return data.puzzle.toLowerCase().split("");
+  } else {
+    throw new Error("unable to get puzzle");
+  }
+}
+
+//단어 받아와서 길이 확인 후 단어 설정하기
+async function setWord() {
+  let word = await getWord(); //단어 받아오기
+  while (word.length != 7) {
```

Comment:
```suggestion
const gameWord = await getWord();
while(word.length !== 7)
```
자바스크립트에서 함수 사용할때는 let변수를 지양하는 게 조아! 
왜냐면 안그래도 유연한 언어라서 쉽게 변경되는 값이 많을수록 오류가 많이 날 수도 있고든,,,!


요렇게 함수 안에서만 사용하고 리턴 하는 거니까 const가 더 적절하고,,!
!== 요걸로 써주구!

Reply:
[ERROR] Cannot assign to "word" because it is a constant

    main.js:59:4:
      59 │     word = await getWord();
         ╵     ~~~~

  The symbol "word" was declared a constant here:

    main.js:57:8:
      57 │   const word = await getWord(); //단어 받아오기
         ╵         ~~~~

Build failed with 1 error:
main.js:59:4: ERROR: Cannot assign to "word" because it is a constant

에러나요오....ㅠㅠ